packages/react-form-state/src/test/FormState.test.tsx

```diff
@@ -652,8 +682,75 @@ describe('<FormState />', () => {
       expect(descriptionError.message).toEqual(submitErrors[1].message);
     });
   });
+
+  describe('performance', () => {
+    it('does not re-render form when state after onchange is identical', () => {
+      const product = faker.commerce.productName();
+      const description = faker.lorem.words();
+
+      const renderPropSpy = jest.fn(() => null);
+
+      mount(
+        <FormState initialValues={{product, description}}>
+          {renderPropSpy}
+        </FormState>,
+      );
+
+      const formDetails = lastCallArgs(renderPropSpy);
+      formDetails.fields.product.onChange(product);
```

Comment:
I think it would be nicer in the tests to have them simulate things a little better, so spread this field into an input and then trigger that input `onChange`

Reply:
probably we actually want both cases